docs/technical/security.md

```diff
@@ -0,0 +1,201 @@
+
+### Specify 'Read Only' access for containers where possible
+
+```yaml
+  containers:
+  - name: <container name>
+    image: <image>
+    securityContext:
+      readOnlyRootFilesystem: true
+```
+
+> We may need to experiment with this one to ensure .NET containers can be run read only.
+
+## Deployed Infrastructure
+
+At time of writing the Buerokratt project will deploy it's resources to Azure using Terraform.
+
+### Complete Separation Between Development and Production Environments
+
+Each deployed environment should have separate resources from one another.  This goes back as far as the Service Principal used to deploy these environments.  (This project uses Service Principal identities to perform infrastructure and application deployment via Terraform).  
+
+No resources should be shared across environments.  e.g. No resources in the 'Dev' environment should be used in the 'Integration' environment.
+
+### Secret Management
+
+All secrets used within this project should be stored using secure stored with deployment Service Principals (for performing deployments) and Service Administrators.
+
+Secrets pertain to data connection strings, access keys or credentials.
+
+GitHub Secrets and Azure Key Vault both offer this capability and can be selected based on the use case.
+
+### vLANs for Resource Protection
+
+Where possible - we can ensure that traffic remains 'hidden' from the internet by creating Virtual Lans.  This project intends to use Cosmos DB as a datastore for the CentOps Service.  
+
+Communication between AKS and CosmosDB can be performed across a vLAN for best protection for this traffic.
+
+### Mutual TLS
+
+Service to service communication within Kubernetes should be protected with mTLS.  This ensures that traffic within the cluster is encrypted, but both client and service authenticate with one another. (See [Mutual TLS](https://en.wikipedia.org/wiki/Mutual_authentication#mTLS))
+
+### IP Filtering on Public Endpoints
+
+Lock down IPs to specific endpoints which are allowed access to particular APIs.
+
+This project will be exposing 'public' and 'private' endpoints from the same clusters.
+
+We should consider opportunities to only allowing access to private endpoints from particular IP addresses. e.g. The Admin API for CentOps can only be called from the network the CentOps Administrator will access it from.
+
+## Security in Operation
+
+### Communication
+
+* HTTPS Everywhere.
+* TLS 1.2 at a bare minimum.
+* Strict CORS policies where applicable.
+
+### Versioning
+
+Developed Components should identify the version of the code they represent.  This allows participants in the Beurokratt system to gate interaction to other components based on their version.  On API endpoints this might be a simple X-Byk-Version header that ties that component to a particular build.

```

Comment:
```suggestion
Developed Components should identify the version of the code they represent.  This allows participants in the Buerokratt system to gate interaction to other components based on their version.  On API endpoints this might be a simple X-Byk-Version header that ties that component to a particular build.

```